website/docs/bestpractices/analytics/spark-best-practices.md

```diff
@@ -0,0 +1,156 @@
+
+#### Consolidation Configuration
+While enabling `consolidation` for Spark executor pods can lead to better cluster resource utilization, it's crucial to strike a balance with job performance. Frequent consolidation events can result in slower execution times for Spark jobs, as executors are forced to recompute the shuffle data and RDD blocks.
+
+This impact is particularly noticeable in long-running Spark jobs. To mitigate this, it's essential to carefully tune the consolidation interval.
+
+To save intermediate data computed in Spark Executors, consider one of the following options, which are discussed in `Storage section`:
+    * PVC attach
+    * External Shuffle storage
+    * Executor handling (Related to handling interruptions)
+        * `spark.executor.decommission.enabled=true`
+        * `spark.storage.decommission.enabled=true`
+
+#### Handling interruptions during Karpenter Consolidation/Spot Termination
+
+Perform a controlled decommissioning instead of killing an executor. Leverage `TerminationGracePeriod`, be executor aware and save shuffle data before node being decommissioned.
+
+Spark also provides native configurations, to control the termination behaviour.
+        * Controlling the `executor` interruptions
+            * Config:
+                * `spark.executor.decommission.enabled`
+                * `spark.executor.decommission.forceKillTimeout`
+            * This is particularly useful in scenarios where executors might be terminated due to spot instance interruptions or Karpenter consolidation events.
+            * This will allow executor to gracefully shutdown. Executors will stop accepting tasks and would notify driver about decommissioning state.
+        * To control executor’s `BlockManager` behaviour
+            * Config:
+                * `spark.storage.decommission.enabled`
+                * `spark.storage.decommission.shuffleBlocks.enabled`
+                * `spark.storage.decommission.rddBlocks.enabled`
+                * `spark.storage.decommission.fallbackStorage.path`
+
+This enables the migration of shuffle and RDD blocks from decommissioning executor to other available executors or to a fallback storage location if configured. This helps in dynamic environment by reducing the need to recompute the shuffle data or RDD blocks, there by improving the job completion times and resource efficiency.
+
+
+## Advanced Scheduling Considerations
+### Default Kubernetes Scheduler behaviour.
+
+Default kubernetes scheduler uses `least allocated` approach. This strategy aims to distribute pods evenly across cluster, which helps in maintaining availability and a balanced resource utilization across all nodes, rather than packing more pods in fewer nodes.
```

Comment:
```suggestion
Default Kubernetes scheduler uses `least allocated` approach. This strategy aims to distribute pods evenly across cluster, which helps in maintaining availability and a balanced resource utilization across all nodes, rather than packing more pods in fewer nodes.
```